### PR TITLE
add WhatsApp media

### PR DESCRIPTION
**Pull request by hughhopkins**
# Description

OAS3 for WhatsApp media on Messages and Dispatch APIs

### CODE REVIEW

File: definitions/messages-olympus.yml [hunk 1]
```diff
@@ -485,15 +490,36 @@ components:
...
+      properties:
+        url:
+          type: string
+          description: 'The URL of the video attachment. `messenger` and `whatsapp` supports .mp4.'
+          minLength: 1
+          maxLength: 2000
+          example: 'https://example.com/video.mp4'
+        caption:
+          type: string
+          description: 'Additional text to accompany the image. Only supported by WhatsApp.'
```
Comment (peterjurkovic) [inline definitions/messages-olympus.yml]:
```suggestion
          description: 'Additional text to accompany the image. Only supported by WhatsApp. Optional. Only present if specified.'
````

File: definitions/messages-olympus.yml [hunk 2]
```diff
@@ -485,15 +490,36 @@ components:
...
         url:
           type: string
-          description: 'The URL of the file attachment. `messenger` supports a wide range of attachments including .zip, .csv and .pdf'
+          description: 'The URL of the file attachment. `messenger` supports a wide range of attachments including .zip, .csv and .pdf. `whatsapp` supports .pdf, .doc(x), .ppt(x) and .xls(x).'
           minLength: 1
           maxLength: 2000
           example: 'https://example.com/file.zip'
+        caption:
+          type: string
+          description: 'Additional text to accompany the image. Only supported by WhatsApp.'
```
Comment (peterjurkovic) [inline definitions/messages-olympus.yml]:
```suggestion
          description: 'Additional text to accompany the image. Only supported by WhatsApp. Optional. Only present if specified.'
```

File: definitions/messages-olympus.yml [hunk 3]
```diff
@@ -248,7 +265,7 @@ components:
               properties:
                 type:
                   type: string
-                  description: The type of message being received.
+                  description: The type of message being received. `whatsapp` and `messenger` supports `text`, `image`, `audio`, `file` and `location`. `mms` supports `image`. `whatsapp` maximum inbound size is 64mb.
```
[the remark this answers is not in the document]
Comment (hughhopkins) [reply definitions/messages-olympus.yml]: updated to the bottom style

File: definitions/messages-olympus.yml [hunk 4]
```diff
@@ -642,6 +686,9 @@ x-errors:
 
   1330:
     description: Unknown  -  An unknown error was received from the carrier who tried to send this this message. Depending on the carrier, that to is unknown. When you see this error, and status is rejected, always check if to in your request was valid.
+  
+  1331:
+    description: Provider error.
```
[the remark this answers is not in the document]
Comment (hughhopkins) [reply definitions/messages-olympus.yml]: add better error

**Pull request merged**